Make Basic Usage tutorial more detail. (#764)
scikit-build/scikit-build: docs/usage.rst

Example of setup.py, CMakeLists.txt and pyproject.toml
------------------------------------------------------
The full example code is `Here <https://github.com/scikit-build/scikit-build-sample-projects/tree/master/projects/hello-cpp>`_

> Make a fold name my_project as your project root folder, place the following in your project's

Reviewer: ```suggestion
Create a directory named ``my_project`` as your project root folder, then add the following in your project's
```

``setup.py`` file::

    from skbuild import setup  # This line replaces 'from setuptools import setup'
    setup(
        name="hello-cpp",
        version="1.2.3",
        description="a minimal example package (cpp version)",
        author='The scikit-build team',
        license="MIT",
        packages=['hello'],
        python_requires=">=3.7",
    )

Your project now uses scikit-build instead of setuptools.

Next, add a ``CMakeLists.txt`` to describe how to build your extension. In the following example,
a C++ extension named ``_hello`` is built::

>     cmake_minimum_required(VERSION 3.4...3.22)

Reviewer: This contradicts the requirement specified below `cmake>=3.18`

>     cmake_minimum_required(VERSION 3.4...3.22)

Reviewer: ```suggestion
    cmake_minimum_required(VERSION 3.18.6)
```

    [build-system]
    requires = [
        "setuptools>=42",
        "scikit-build>=0.13",
        "cmake>=3.18",

>         "ninja",

Reviewer: ```suggestion
        "ninja; platform_system!='Windows'",
```

This contradicts the comment below and should be maintained.

    ]
    build-backend = "setuptools.build_meta"

> Make a hello folder inside my_project folder and place `_hello.cxx <https://github.com/scikit-build/scikit-build-sample-projects/blob/8fdbc8a0dd78656ea0b431e005b49f3e19786444/projects/hello-cpp/hello/_hello.cxx>`_ and `__init__.py <https://github.com/scikit-build/scikit-build-sample-projects/blob/8fdbc8a0dd78656ea0b431e005b49f3e19786444/projects/hello-cpp/hello/__init__.py>`_ inside hello folder.

Reviewer: ```suggestion
Create a ``hello`` directory inside ``my_project`` root directory folder and place the files `_hello.cxx <https://github.com/scikit-build/scikit-build-sample-projects/blob/8fdbc8a0dd78656ea0b431e005b49f3e19786444/projects/hello-cpp/hello/_hello.cxx>`_ and `__init__.py <https://github.com/scikit-build/scikit-build-sample-projects/blob/8fdbc8a0dd78656ea0b431e005b49f3e19786444/projects/hello-cpp/hello/__init__.py>`_ inside the ``hello`` directory.
```

> Now every thing is ready, go to my_project's parent folder and type following command to install your extension::

Reviewer: ```suggestion
Now every thing is ready, go to my_project's parent directory and type the following command to install your extension::
```
